src/pages/post/2024-04-23-TensorRT-LLM.mdx

```diff
@@ -0,0 +1,275 @@
+- TensorRT-LLM is up to **30% faster** on laptop GPUs (e.g. 4070 GPUs) with smaller VRAM
+
+![image](./_assets/throughput_benchmark.png)
+
+| Throughput (Higher is Better)                           | TensorRT-LLM    | Llama.cpp    | % Difference    |
+| -------------------------- | --------------- | ------------ | --- |
+| GeForce RTX 4090 desktop GPU           | ✅ 170.63t/s     | 100.43t/s     | 69.89% faster    |
+| GeForce RTX 3090 desktop GPU           | ✅ 144.9t/s    | 88.67t/s        | 62.57% faster   |
+| GeForce RTX 4070 Laptop GPU           | ✅ 51.57t/s     | 39.7t/s     | 29.9% faster   |
+| GeForce RTX 4090 eGPU           | ✅ 104.95t/s    | 62.22t/s     | 68.66% faster   |
+| Laptop Intel i7 13800H CPU | (Not supported) | ✅ 11.566t/s |     |
+
+### Max VRAM Utilization
+- TensorRT-LLM used marginally more average VRAM utilization at peak utilization vs. llama.cpp (up to 14%). Though this could have interesting implications on consuming more electricity over time.
+- Note: we used comparable (but not identical) quantizations, and TensorRT-LLM’s `AWQ INT4` is implemented differently from llama.cpp’s `q4_k_m`
+
+|  Average VRAM utilization %                         | TensorRT-LLM    | Llama.cpp    | % Difference    |
+| -------------------------- | --------------- | ------------ | --- |
+| GeForce RTX 4090 desktop GPU           | 72.10      |  64.00     | 12.66% more    |
+| GeForce RTX 3090 desktop GPU           | 76.20     | 66.80        | 14.07% more    |
+| GeForce RTX 4070 Laptop GPU           | 81.22     | 72.78     |  11.06% more      |
+| GeForce RTX 4090 eGPU           | N/A     | N/A     | N/A  |
+
+### Max RAM Usage
+TensorRT-LLM uses a lot less Max RAM vs. llama.cpp on slower connection (PCIe 3.0 or Thunderbolt 3) due to better memory management and efficient delegation to VRAM. On faster connection, it’s at least equal to llama.cpp.
+
+|  Max RAM utilization                    | TensorRT-LLM    | Llama.cpp    | % Difference    |
+| -------------------------- | --------------- | ------------ | --- |
+| GeForce RTX 4090 desktop GPU           |   ✅ 4.98      |  7.11    |  ✅ 29.88% less  |
+| GeForce RTX 3090 desktop GPU           |   ✅ 0.98      |  2.60    |   ✅ 62.41% less  |
+| GeForce RTX 4070 Laptop GPU          |   ✅ 1.04    |  4.44    | ✅  76.55%% less     |
+| GeForce RTX 4090 eGPU           | ✅ 4.11    |  5.38    | ✅ 23.61% less    |
+
+### Compiled Model Size and Number of files
+- Contrary to popular belief, TensorRT-LLM prebuilt models turned out to not be that huge
+Mistral 7b int4 was actually **25% smaller** in TensorRT-LLM, at 3.05gb vs. 4.06gb
+- Note: These are approximate comparisons, as TensorRT-LLM’s `AWQ INT4` is implemented differently from llama.cpp’s `q4_k_m`
+- The bigger takeaway is that the compiled model sizes are roughly in the same ballpark, while the number of files for TensorRT-LLM is 7x the GGUF number of files.
```

Comment:
`The bigger takeaway is that the Compiled model sizes are roughly in the same ballpark, while the number of files for TensorRT-LLM is 7x the GGUF number of files.`